### ISSUE TITLE
`swift catalog search` should support more complex queries

### ISSUE THREAD

**Issue by helje5** [kiliankoe/apodidae#42]
Currently you can only search repo strings:

> The only extra logic is that forks are searched as well if a query in the form of a github slug, e.g. ZeeQL/ZeeQL3 is given

E.g. I would like to query SPM packages which are 10*+, order by stars.
Another useful query would be "packages referred to by N+ other packages" (popular by reference, though that may need to imply indirection).

**Comment by kiliankoe** [comment]
good idea! I'm not entirely sure how much is possible through GH's API, but it's more than worth the investigation!
